packages/backend/src/models/DashboardModel/DashboardModel.ts

```diff
@@ -65,57 +70,67 @@ export class DashboardModel {
+        version.tiles.forEach((tile) => {
+            const { uuid: dashboardTileId, type, w, h, x, y } = tile;
+            promises.push(
+                (async () => {
+                    const [insertedTile] = await trx(DashboardTilesTableName)
+                        .insert({
+                            dashboard_version_id:
+                                versionId.dashboard_version_id,
+                            dashboard_tile_uuid: dashboardTileId,
+                            type,
+                            height: h,
+                            width: w,
+                            x_offset: x,
+                            y_offset: y,
+                        })
+                        .returning('*');
+                    if (
+                        tile.type === DashboardTileTypes.SAVED_CHART &&
+                        tile.properties.savedChartUuid
+                    ) {
+                        const [savedChart] = await trx(SavedQueriesTableName)
+                            .select(['saved_query_id'])
+                            .where(
+                                'saved_query_uuid',
+                                tile.properties.savedChartUuid,
                             )
-                                .select(['saved_query_id'])
-                                .where(
-                                    'saved_query_uuid',
-                                    properties.savedChartUuid,
-                                )
-                                .limit(1);
-                            if (!savedChart) {
-                                throw new NotFoundError(
-                                    'Saved chart not found',
-                                );
-                            }
-                            await trx(DashboardTileChartTableName).insert({
-                                dashboard_version_id:
-                                    versionId.dashboard_version_id,
-                                dashboard_tile_uuid:
-                                    insertedTile.dashboard_tile_uuid,
-                                saved_chart_id: savedChart.saved_query_id,
-                            });
+                            .limit(1);
+                        if (!savedChart) {
+                            throw new NotFoundError('Saved chart not found');
                         }
-                    })(),
-                );
-            },
-        );
+                        await trx(DashboardTileChartTableName).insert({
+                            dashboard_version_id:
+                                versionId.dashboard_version_id,
+                            dashboard_tile_uuid:
+                                insertedTile.dashboard_tile_uuid,
+                            saved_chart_id: savedChart.saved_query_id,
+                        });
+                    }
+                    if (tile.type === DashboardTileTypes.MARKDOWN) {
```

Comment:
Can we use a switch to catch the default case? Would help if code / migrations get out of sync